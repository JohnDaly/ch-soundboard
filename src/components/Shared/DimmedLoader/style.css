@keyframes indeterminate {
	0% {
		left: -35%;
		right: 100%
	}

	100%, 60% {
		left: 100%;
		right: -90%
	}
}

@keyframes indeterminate-short {
	0% {
		left: -200%;
		right: 100%
	}

	100%, 60% {
		left: 107%;
		right: -8%
	}
}

.progress {
	position: relative;
}

.progress-xs,
.progress-xs .progress-bar {
	height: .25rem;
}

.progress-sm,
.progress-sm .progress-bar {
	height: .5rem;
}

.progress-bar-indeterminate:before {
    content: '';
    position: absolute;
    background-color: inherit;
    left: 0;
    will-change: left, right;
    top: 0;
    bottom: 0;
    animation: indeterminate 2.1s cubic-bezier(.65, .815, .735, .395) infinite;
}

.progress-bar-indeterminate:after {
    content: '';
    position: absolute;
    background-color: inherit;
    left: 0;
    will-change: left, right;
    top: 0;
    bottom: 0;
    animation: indeterminate-short 2.1s cubic-bezier(.165, .84, .44, 1) infinite;
	animation-delay: 1.15s;
}



@keyframes loader {
	from {
		transform: rotate(0deg);
	}

	to {
		transform: rotate(360deg);
	}
}


/**
Dimmer
*/
.dimmer {
    position: relative;
}

.dimmer .loader {
    display: block;
    opacity: 0;
    z-index: 0;
    margin: 0 auto;
    position: absolute;
    top: 50%;
    left: 0;
    right: 0;
    transform: translateY(-50%);
}

.dimmer.active .loader {
    display: block;
    opacity: 1;
    z-index: 10;
}

.dimmer.active .dimmer-content {
    opacity: .5;
	pointer-events: none;
}

/**
Loader
*/
.loader {
    transition: all 0.5s ease;
    display: block;
    position: relative;
    height: 2.5rem;
    width: 2.5rem;
    z-index: 10;
}

.loader:before {
    width: 2.5rem;
    height: 2.5rem;
    margin: -1.25rem 0 0 -1.25rem;
    position: absolute;
    content: '';
    top: 50%;
    left: 50%;

    border-radius: 50%;
	border: 3px solid currentColor;
	opacity: .15;
}

.loader:after {
    width: 2.5rem;
    height: 2.5rem;
    margin: -1.25rem 0 0 -1.25rem;
    position: absolute;
    content: '';
    top: 50%;
    left: 50%;

    animation: loader .6s linear;
    animation-iteration-count: infinite;

    border-radius: 50%;
    border: 3px solid;
    border-color: transparent;
    border-top-color: currentColor;
    box-shadow: 0 0 0 1px transparent;
}