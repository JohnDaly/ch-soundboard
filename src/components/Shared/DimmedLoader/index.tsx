// tslint:disable

// External Dependencies
import * as React from 'react'

// Internal Dependencies
import './style.css'

interface ComponentProps {
	isLoading: boolean
	component: any
}

type Props = ComponentProps

export default class DimmedLoader extends React.Component<Props> {
	constructor(props: Props) {
		super(props)
	}

	public render() {
		if (!this.props.isLoading) {
            return (
                <div className='dimmer h-100'>
                    <div className='loader'/>
                    <div className='dimmer-content'>
                        {this.props.component}
                    </div>
                </div>
            )
		}

		return (
			<div className='dimmer active h-100'>
				<div className='loader'/>
				<div className='dimmer-content'>
					{this.props.component}
				</div>
			</div>
		)
	}
}
