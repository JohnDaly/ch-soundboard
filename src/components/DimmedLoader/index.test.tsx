// External Dependencies
import { mount, shallow } from 'enzyme'
import enzymeToJSON from 'enzyme-to-json'
import 'jest-styled-components'
import React from 'react'

// Internal Dependencies
import DimmedLoader from './index'

describe('DimmedLoader', () => {
    it('renders without crashing', () => {
        const wrapper = mount(
            <DimmedLoader isLoading={true}>
                <h1>Hello, world!</h1>
            </DimmedLoader>
        )
        expect(wrapper).toBeDefined()
    })

    it('should match the snapshot (isLoading = true)', () => {
        const shallowWrapper = shallow(
            <DimmedLoader isLoading={true}>
                <h1>Hello, world!</h1>
            </DimmedLoader>
        )
        expect(enzymeToJSON(shallowWrapper)).toMatchSnapshot()
    })

    it('should match the snapshot (isLoading = false)', () => {
        const shallowWrapper = shallow(
            <DimmedLoader isLoading={false}>
                <h1>Hello, world!</h1>
            </DimmedLoader>
        )
        expect(enzymeToJSON(shallowWrapper)).toMatchSnapshot()
    })
})