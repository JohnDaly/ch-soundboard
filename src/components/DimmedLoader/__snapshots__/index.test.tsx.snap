// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DimmedLoader should match the snapshot (isLoading = false) 1`] = `
<Dimmer>
  <Loader
    active={false}
  />
  <DimmerContent
    active={false}
  >
    <h1>
      Hello, world!
    </h1>
  </DimmerContent>
</Dimmer>
`;

exports[`DimmedLoader should match the snapshot (isLoading = true) 1`] = `
<Dimmer>
  <Loader
    active={true}
  />
  <DimmerContent
    active={true}
  >
    <h1>
      Hello, world!
    </h1>
  </DimmerContent>
</Dimmer>
`;
